webapi/src/main/scala/org/knora/webapi/store/triplestore/upgrade/RepositoryUpdater.scala

```diff
@@ -97,57 +73,33 @@ final case class RepositoryUpdater(triplestoreService: TriplestoreService) {
    *
    * @return the `knora-base` version string, if any, in the repository.
    */
-  private def getRepositoryVersion(): Task[Option[String]] =
-    for {
-      repositoryVersionResponse <- triplestoreService.query(Select(knoraBaseVersionQuery, isGravsearch = false))
-      bindings                  <- ZIO.succeed(repositoryVersionResponse.results.bindings)
-      versionString <-
-        if (bindings.nonEmpty) {
-          ZIO.succeed(Some(bindings.head.rowMap("knoraBaseVersion")))
-        } else {
-          ZIO.none
-        }
-    } yield versionString
+  private def getRepositoryVersion: Task[Option[String]] =
+    triplestoreService
+      .query(
+        Select(
+          """PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+            |
+            |SELECT ?knoraBaseVersion WHERE {
+            |    <http://www.knora.org/ontology/knora-base> knora-base:ontologyVersion ?knoraBaseVersion .
+            |}""".stripMargin,
+        ),
+      )
+      .map(_.results.bindings.headOption.map(_.rowMap("knoraBaseVersion")))
 
   /**
    * Constructs a list of update plugins that need to be run to update the repository.
    *
-   * @param maybeRepositoryVersionString the `knora-base` version string, if any, in the repository.
+   * @param maybeRepositoryVersion the `knora-base` version string, if any, in the repository.
    * @return the plugins needed to update the repository.
    */
-  private def selectPluginsForNeededUpdates(
-    maybeRepositoryVersionString: Option[String],
-  ): UIO[Seq[PluginForKnoraBaseVersion]] = {
-
-    // A list of available plugins.
-    val plugins: Seq[PluginForKnoraBaseVersion] =
-      RepositoryUpdatePlan.makePluginsForVersions(log)
-
-    ZIO.attempt {
-      maybeRepositoryVersionString match {
-        case Some(repositoryVersion) =>
-          // The repository has a version string. Get the plugins for all subsequent versions.
-
-          // Make a map of version strings to plugins.
-          val versionsToPluginsMap: Map[String, PluginForKnoraBaseVersion] = plugins.map { plugin =>
-            s"knora-base v${plugin.versionNumber}" -> plugin
-          }.toMap
-
-          val pluginForRepositoryVersion: PluginForKnoraBaseVersion =
-            versionsToPluginsMap.getOrElse(
-              repositoryVersion,
-              throw InconsistentRepositoryDataException(s"No such repository version $repositoryVersion"),
-            )
-
-          plugins.filter { plugin =>
-            plugin.versionNumber > pluginForRepositoryVersion.versionNumber
-          }
-
-        case None =>
-          // The repository has no version string. Include all updates.
-          plugins
-      }
-    }.orDie
+  private def selectPluginsForNeededUpdates(maybeRepositoryVersion: Option[String]): Seq[PluginForKnoraBaseVersion] = {
+    val repositoryVersion = maybeRepositoryVersion.flatMap(_.replace("knora-base v", "").toIntOption).getOrElse(-1)
```

Comment:
so if this is not found, then all plugins are executed?

Reply:
Yes, if the db does not have a valid version we run all migrations.